nomad/client_agent_endpoint.go

```diff
@@ -236,3 +255,60 @@ func (m *Agent) forwardMonitor(conn io.ReadWriteCloser, args cstructs.MonitorReq
 	structs.Bridge(conn, clientConn)
 	return
 }
+
+func (m *Agent) forwardMonitorServer(conn io.ReadWriteCloser, args cstructs.MonitorRequest, encoder *codec.Encoder, decoder *codec.Decoder) {
+	var target *serverParts
+	serverID := args.ServerID
+
+	// empty ServerID to prevent forwarding loop
+	args.ServerID = ""
+
+	if serverID == "leader" {
+		isLeader, remoteServer := m.srv.getLeader()
+		if !isLeader && remoteServer != nil {
+			target = remoteServer
+		}
+		if !isLeader && remoteServer == nil {
+			err := fmt.Errorf("no leader")
```

Comment:
Use this error: https://github.com/hashicorp/nomad/blob/c24a631defb3269191646a4e59310261db8de3a3/nomad/structs/errors.go#L31